bin/metrics-es-node-graphite.rb

```diff
@@ -264,7 +264,9 @@ def run
     node['indices'].each do |type, index|
       index.each do |k, v|
         # #YELLOW
-        unless k =~ /(_time$)/ || v =~ /\d+/
+        if k.end_with? 'is_throttled'
+          metrics["indices.#{type}.#{k}"] = true?(v) ? 1 : 0
```

Comment:
The only thing I am not sure on is what is the best thing to do here. I am leaning towards:
```
case true?(v)
when true
  1
when false
  0
else
  # not sure if we should raise, ok, critical, or return a `(0|1)`
  raise(true?(v))
end
```

I would not worry about implementing it until I have had someone else take a look.

Reply:
Hmm does it make sense to have `true?` raise in the:
```
  else
    "#{obj} is not a truthy value, please open an issue with this output so we can fix it"
```
bit?

Reply:
I thought about that initially but was not sure what we wanted to do so I illustrated what **could** be done and this allowed us to always `return` something and then the handling of that could be implemented independently.

Reply:
based on the feedback and what I was already leaning towards I think the correct solution would be to do something like:
```
else
    STDERR.puts "#{obj} is not a truthy value, please open an issue with this output so we can fix it"
```

Reply:
Ah I see, I see. So off hand, I'm not aware of a good way for a metric script to actually log to sensu (although the STDERR bit you mentioned might work, just worried that'll also get sent along). 

Unless there's something I'm not aware of, I'd lean towards marking the current metric as unknown if possible otherwise, let's catch and `raise` here. I _hope_ this condition is rare enough that this code path never has to happen.

Reply:
FWIW, `is_throttled` also feels like something that might work better as a dedicated Sensu check outside of this metric script.

Reply:
Assuming the change to have it print to `STDERR` and our assumptions and understanding of the code are correct I think the logic here would look something like this:
```
if k.end_with? 'is_throttled'
  state = true?(v)
  unless state.class == 'String'
    metrics["indices.#{type}.#{k}"] = true?(v) ? 1 : 0
  end
end
```

that would return the right value when it can and when it can't it should return nothing but print a useful message to `STDERR` that sensu-client will pick up. I would think that any handler would not be affected as I would be surprised if sensu client sent it as `STDIN` which is typically what a pipe handler will by default work with. I would love some clarification from someone at sensu.

Reply:
> FWIW, `is_throttled` also feels like something that might work better as a dedicated Sensu check outside of this metric script.

I can see that but IMHO they are separate concerns, I tend to try to conceptualize all sensu "checks" into the following broad categories:
- state checking: this can be something against a local host, a remote host, etc and can either be looking at the current state of a system or some kind of representation of previous states. To further clarify you could search against a TSDB to get the average, min, max, standard deviation, etc of the last `n` number of state results. This is what I internally refer to as metric checks but avoid this terminology externally because it can confuse people.
- data collection: these are what are commonly referred to the "metric" scripts in the sensu community. I think they should attempt to report data, bail with an error, or potentially a combination.

I think that yes it makes a lot of sensu to create a "state check" for that but the question is do you want to check against the current state or against something else.

Anyways those are just some thoughts/opinions and I doubt there is a "right" or "wrong" answer and its a matter of perspective. Perhaps we make this a CLI arg to control this?

Reply:
Actually I thought about it and I think we should return a `String` or a `(True|False)Class` object and do the handling in the other function:
```ruby
if k.end_with? 'is_throttled'
  state = true?(v)
  if state.class == 'String'
    STDERR.puts state
   # optionally we could additionally set the value to `nil` here but that relies on the appropriate system from handling this properly which seems less than ideal
  # to play devils advocate though we could have something at the end of doing all the metric massinging drop any metrics with a `nil` value which I think does have some real value as it means that we can do a final cleanup/sanitization of the output before we ship it.
  else
    metrics["indices.#{type}.#{k}"] = true?(v) ? 1 : 0
  end
end
```

I'm still not confident that this is the correct approach either but its something to consider.